src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -990,6 +1007,16 @@ enum Disambiguator {
 }
 
 impl Disambiguator {
+    /// The text that should be displayed when the link is rendered as HTML
+    fn display_for(&self, link: &str) -> String {
+        match self {
+            // FIXME: this will have different output if the user had `m!()` originally.
+            Self::Kind(DefKind::Macro(MacroKind::Bang)) => format!("{}!", link),
```

Comment:
Why not just show everything in link before and including `!`?
But I don't know why do we need to change the text to be rendered in HTML. Shouldn't we use the original text?

Reply:
`link` is a bad name here: it should say `path` instead. The entire issue is about _not_ using the original text, because it looks bad for e.g. `[mod@char]`.